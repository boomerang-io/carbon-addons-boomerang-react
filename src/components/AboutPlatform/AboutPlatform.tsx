/*
IBM Confidential
694970X, 69497O0
Â© Copyright IBM Corp. 2022, 2024
*/


import React from "react";
import { ComposedModal, ModalHeader, ModalBody } from "@carbon/react";
import { Information } from "@carbon/react/icons";
import HeaderMenuItem from "../Header/HeaderMenuItem";
import IBMCloudIcon from "./assets/IBMCloudIcon";
import KubernetesIcon from "./assets/KubernetesIcon";
import MongoDbIcon from "./assets/MongoDbIcon";
import OpenShiftIcon from "./assets/OpenShiftIcon";
import ReactIcon from "./assets/ReactIcon";
import SpringIcon from "./assets/SpringIcon";
import { prefix } from "../../internal/settings";

const iconClassName = `${prefix}--bmrg-aboutPlatform-images__img`;

type Props = {
  closeModal: () => void;
  isOpen: boolean;
  name: string;
  version: string;
};

function AboutPlatform({ closeModal, isOpen = false, version, name }: Props) {
  return (
    <ComposedModal
      open={isOpen}
      className={`${prefix}--bmrg-aboutPlatform-container ${prefix}--bmrg-header-modal`}
      onClose={closeModal}
    >
      <ModalHeader label={`${name}  |  Version ${version}`} title="About the Platform" closeModal={closeModal} />
      <ModalBody>
        <footer className={`${prefix}--bmrg-aboutPlatform-footer`}>
          <h1 className={`${prefix}--bmrg-aboutPlatform-footer__header`}>Powered by</h1>
          <ul className={`${prefix}--bmrg-aboutPlatform-images`}>
            <li key="ibm-cloud-icon">
              <a href="https://www.ibm.com/cloud" target="_blank" rel="noopener noreferrer" title="IBM Cloud">
                <IBMCloudIcon className={iconClassName} />
              </a>
            </li>
            <li key="openshift-icon">
              <a
                href="https://www.redhat.com/en/technologies/cloud-computing/openshift"
                target="_blank"
                rel="noopener noreferrer"
                title="OpenShift"
              >
                <OpenShiftIcon className={iconClassName} />
              </a>
            </li>
            <li key="kubernetes-icon">
              <a href="https://kubernetes.io/" target="_blank" rel="noopener noreferrer" title="Kubernetes">
                <KubernetesIcon className={iconClassName} />
              </a>
            </li>
            <li key="spring-icon">
              <a href="https://spring.io/" target="_blank" rel="noopener noreferrer" title="Spring">
                <SpringIcon className={iconClassName} />
              </a>
            </li>
            <li key="react-icon">
              <a href="https://reactjs.org/" target="_blank" rel="noopener noreferrer" title="React">
                <ReactIcon className={iconClassName} />
              </a>
            </li>
            <li key="mongodb-icon">
              <a href="https://www.mongodb.com/" target="_blank" rel="noopener noreferrer" title="MongoDB">
                <MongoDbIcon className={iconClassName} />
              </a>
            </li>
          </ul>
        </footer>
      </ModalBody>
    </ComposedModal>
  );
}

export default AboutPlatform;

function AboutPlatformMenuItem(props: Omit<Props, "isOpen" | "closeModal">) {
  const menuItemRef = React.useRef<HTMLLinkElement>(null);
  const [isOpen, setIsOpen] = React.useState(false);

  const handleClose = () => {
    setIsOpen(false);
    setTimeout(() => {
      menuItemRef.current?.focus();
    }, 0);
  };

  return (
    <>
      <HeaderMenuItem
        icon={<Information />}
        onClick={() => setIsOpen(!isOpen)}
        ref={menuItemRef}
        text="About Platform"
        type="button"
      />
      <AboutPlatform isOpen={isOpen} closeModal={handleClose} {...props} />
    </>
  );
}

export { AboutPlatformMenuItem };
