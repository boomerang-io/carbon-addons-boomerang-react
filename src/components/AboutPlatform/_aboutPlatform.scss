@use "sass:map";
@use "../../styles/config" as *;
@use "../../styles/themes/shell-tokens" as shell;
@use "../../styles/utils";

.#{$prefix}--bmrg-aboutPlatform-container {
  .#{$prefix}--modal-container {
    height: utils.rem(244px);
    width: utils.rem(500px);
  }

  .#{$prefix}--modal-content {
    height: 100%;
  }

  .#{$prefix}--modal-header {
    .#{$prefix}--modal-header__label.#{$prefix}--type-delta {
      white-space: pre;
    }
  }

  .#{$prefix}--bmrg-aboutPlatform-footer__header {
    color: shell.$bmrg-brand-white;
    margin-bottom: utils.rem(18px);
    margin-top: utils.rem(16px);
    font-family: "IBM Plex Sans", "Helvetica Neue", Arial, Helvetica, sans-serif;
    font-size: utils.rem(14px);
    font-weight: 500;
    line-height: utils.rem(18px);
  }

  .#{$prefix}--bmrg-aboutPlatform-images {
    display: flex;
  }

  .#{$prefix}--bmrg-aboutPlatform-images__img {
    height: 2rem;
    margin-right: 2rem;
    width: auto;
  }
}
