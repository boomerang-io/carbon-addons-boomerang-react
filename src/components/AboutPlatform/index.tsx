export { default, AboutPlatformMenuItem } from "./AboutPlatform";
