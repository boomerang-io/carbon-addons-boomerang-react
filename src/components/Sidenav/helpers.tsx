import React from "react";

export const header = () => <div style={{ fontSize: "5rem", color: "#047cc0" }}>title</div>;
export const SidenavOutstandingTeams = ({ outstandingTeamRequests }: any) => {
  const numberOfOutstandingTeamRequests = outstandingTeamRequests.length;

  if (numberOfOutstandingTeamRequests === 0) {
    return null;
  }

  return (
    <>
      <div
        style={{
          height: "5rem",
          color: "black",
          textAlign: "center",
          marginTop: "1rem",
        }}
        data-tip
        data-for="outstandingTeamRequests"
      >{`${numberOfOutstandingTeamRequests} ${
        numberOfOutstandingTeamRequests === 1 ? "REQUEST" : "REQUESTS"
      } PENDING`}</div>
    </>
  );
};

export const outstandingTeamRequests = [
  {
    id: "59e5720c0aa79f128e639448",
    groupName: "Pending Test Team",
  },
];

export const navItems = [
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: true,
    text: "Demo team 1",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home",
    exact: true,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 21",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home1",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 223",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home2",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 243",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home3",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 254657",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home4",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2345345",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home5",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2334533",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home6",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 24345345345345",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home7",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2334535345345345",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home8",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 256657657653424786783323",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home9",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home10",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 24557423432",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home11",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 12",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home12",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2564",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home13",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 234",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home14",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 288676443",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home15",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2222",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home1",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 25555",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home16",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2666",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home17",
    exact: false,
  },
  {
    dateCreated: "2017-08-28T18:10:11Z",
    id: "59a45c83b0756d1fa0c26ffb",
    isActive: false,
    text: "Demo team 2899",
    ownerEmail: "tester@us.ibm.com",
    ownerId: "59a45e60b0756d1fa0c27825",
    ownerName: "Test user",
    purpose: "Testing",
    path: "/admin/home18",
    exact: false,
  },
];
