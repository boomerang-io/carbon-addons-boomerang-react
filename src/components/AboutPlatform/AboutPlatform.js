import React from 'react';
import PropTypes from 'prop-types';
import { settings } from 'carbon-components';
import { ModalHeader, ModalBody } from 'carbon-components-react/lib/components/ComposedModal';
import IBMCloudIcon from './assets/IBMCloudIcon';
import HeaderMenuItem from '../HeaderMenuItem';
import KubernetesIcon from './assets/KubernetesIcon';
import MongoDbIcon from './assets/MongoDbIcon';
import OpenShiftIcon from './assets/OpenShiftIcon';
import ReactIcon from './assets/ReactIcon';
import SpringIcon from './assets/SpringIcon';

const { prefix } = settings;

const AboutPlatformContainer = ({ version, organization }) => {
  return (
    <HeaderMenuItem
      text="About the Platform"
      iconName="information"
      className={`${prefix}--bmrg-aboutPlatform-container`}
    >
      {({ closeModal }) => {
        return (
          <>
            <ModalHeader
              closeModal={closeModal}
              label={`${organization} ${' '} | ${' '} version ${version}`}
              title="About the Platform"
            />
            <ModalBody>
              <footer className={`${prefix}--bmrg-aboutPlatform-footer`}>
                <h1 className={`${prefix}--bmrg-aboutPlatform-footer__header`}>Powered by</h1>
                <ul className={`${prefix}--bmrg-aboutPlatform-images`}>
                  <li key="ibm-cloud-icon">
                    <IBMCloudIcon
                      alt="IBM Cloud Icon"
                      className={`${prefix}--bmrg-aboutPlatform-images__img`}
                    />
                  </li>
                  <li key="openshift-icon">
                    <OpenShiftIcon
                      alt="Openshift Icon"
                      className={`${prefix}--bmrg-aboutPlatform-images__img`}
                    />
                  </li>
                  <li key="kubernetes-icon">
                    <KubernetesIcon
                      alt="Kubernetes Icon"
                      className={`${prefix}--bmrg-aboutPlatform-images__img`}
                    />
                  </li>
                  <li key="spring-icon">
                    <SpringIcon
                      alt="Spring Icon"
                      className={`${prefix}--bmrg-aboutPlatform-images__img`}
                    />
                  </li>
                  <li key="react-icon">
                    <ReactIcon
                      alt="React Icon"
                      className={`${prefix}--bmrg-aboutPlatform-images__img`}
                    />
                  </li>
                  <li key="mongodb-icon">
                    <MongoDbIcon
                      alt="Mongodb Icon"
                      className={`${prefix}--bmrg-aboutPlatform-images__img`}
                    />
                  </li>
                </ul>
              </footer>
            </ModalBody>
          </>
        );
      }}
    </HeaderMenuItem>
  );
};

AboutPlatformContainer.propTypes = {
  version: PropTypes.string,
  organization: PropTypes.string,
  closeModal: PropTypes.func,
};

AboutPlatformContainer.defaultProps = {
  organization: 'IBM',
  version: '',
};

export default AboutPlatformContainer;
