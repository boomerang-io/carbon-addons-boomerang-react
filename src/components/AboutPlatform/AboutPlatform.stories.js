import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs/react';

import AboutPlatform from './AboutPlatform';

storiesOf('AboutPlatform', module).add('default', () => {
  return (
    <AboutPlatform
      organization={text('organization', 'IBM Boomerang Platform')}
      version={text('version', '5.0.0')}
    />
  );
});
