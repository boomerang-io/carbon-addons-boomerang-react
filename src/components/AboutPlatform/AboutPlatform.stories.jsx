import AboutPlatform from "./AboutPlatform";

export default {
  title: "Platform/AboutPlatform",
  component: AboutPlatform,
  parameters: {
    docs: {
      description: {
        component: "Header menu item that displays modal with key metadata about the platform.",
      },
    },
  },
  argTypes: {
    isFlowApp: { control: "boolean", defaultValue: false },
    organization: { control: "text", defaultValue: "IBM" },
    version: { control: "text", defaultValue: "5.0.0" },
  },
};

export const Default = (args) => {
  return <AboutPlatform {...args} />;
};

export const Flow = (args) => {
  return <AboutPlatform {...args} />;
};

Flow.args = {
  isFlowApp: true,
};
