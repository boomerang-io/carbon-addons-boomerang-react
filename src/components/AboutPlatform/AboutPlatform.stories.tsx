import React from "react";
import AboutPlatform from "./AboutPlatform";
import { headerModalProps } from "../../internal/helpers";

export default {
  title: "Platform/AboutPlatform",
  component: AboutPlatform,
  parameters: {
    docs: {
      description: {
        component: "Header modal with key metadata about the platform.",
      },
    },
  },
};

export const Default = (args) => {
  return <AboutPlatform {...args} />;
};

Default.args = {
  name: "Boomerang",
  version: "1.0.0",
  ...headerModalProps,
};