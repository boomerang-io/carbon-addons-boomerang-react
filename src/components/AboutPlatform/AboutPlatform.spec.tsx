import React from "react";
import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import { axe } from "jest-axe";
import AboutPlatform from "./AboutPlatform";
import { headerModalProps } from "../../internal/helpers";

test("AboutPlatform - snapshot", async () => {
  const { baseElement } = render(<AboutPlatform name="Boomerang" version="1.2.1" {...headerModalProps} />);
  expect(baseElement).toMatchSnapshot();
});

test("AboutPlatform - correct name and version", () => {
  // Render new instance in every test to prevent leaking state
  const { getByText } = render(<AboutPlatform name="Boomerang" version="1.2.1" {...headerModalProps} />);
  const version = getByText(/1.2.1/i);
  const organization = getByText(/Boomerang/i);
  expect(version).toBeInTheDocument();
  expect(organization).toBeInTheDocument();
});

test("AboutPlatform - accessibility", async () => {
  const { container } = render(<AboutPlatform name="Boomerang" version="1.2.1" {...headerModalProps} />);
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});
