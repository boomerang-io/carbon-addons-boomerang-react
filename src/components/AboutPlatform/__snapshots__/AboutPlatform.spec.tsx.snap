// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutPlatform > snapshot 1`] = `
<body>
  <div>
    <div
      aria-hidden="false"
      class="cds--layer-one cds--modal is-visible cds--bmrg-aboutPlatform-container cds--bmrg-header-modal cds--bmrg-aboutPlatform-modalheader"
      role="presentation"
    >
      <div
        aria-modal="true"
        class="cds--modal-container"
        role="dialog"
      >
        <button
          class="cds--visually-hidden"
          type="button"
        >
          Focus sentinel
        </button>
        <div
          class="cds--modal-container-body"
        >
          
          <div
            class="cds--modal-header cds--bmrg-aboutPlatform-modaltitle"
          >
            <h2
              class="cds--modal-header__heading cds--type-beta"
            >
              About the Platform
            </h2>
            <div
              class="cds--modal-close-button"
            >
              <span
                class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--open cds--popover--left cds--tooltip cds--icon-tooltip"
              >
                <div
                  class="cds--tooltip-trigger__wrapper"
                >
                  <button
                    aria-label="Close"
                    aria-labelledby="tooltip-2"
                    class="cds--btn--icon-only cds--modal-close cds--btn cds--btn--primary"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="cds--modal-close__icon"
                      fill="currentColor"
                      focusable="false"
                      height="20"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 32 32"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.4141 16L24 9.4141 22.5859 8 16 14.5859 9.4143 8 8 9.4141 14.5859 16 8 22.5859 9.4143 24 16 17.4141 22.5859 24 24 22.5859 17.4141 16z"
                      />
                    </svg>
                  </button>
                </div>
                <span
                  aria-hidden="false"
                  class="cds--popover"
                  id="tooltip-2"
                  role="tooltip"
                >
                  <span
                    class="cds--popover-content cds--tooltip-content"
                  >
                    Close
                  </span>
                  <span
                    class="cds--popover-caret"
                  />
                </span>
              </span>
            </div>
          </div>
          <div
            class="cds--layer-two cds--modal-content"
          >
            <div>
              <h5
                class="cds--bmrg-aboutPlatform-component-header"
              >
                Components
              </h5>
              <ul>
                <li
                  class="cds--bmrg-aboutPlatform-li-between-first"
                >
                  <span
                    class="cds--bmrg-aboutPlatform-li"
                  >
                    IBM Consulting Advantage
                  </span>
                  <span
                    class="cds--bmrg-aboutPlatform-li-version"
                  >
                    1.2.1
                  </span>
                </li>
                <li
                  class="cds--bmrg-aboutPlatform-li-between"
                >
                  <span
                    class="cds--bmrg-aboutPlatform-li"
                  >
                    Assistants 
                  </span>
                  <span
                    class="cds--bmrg-aboutPlatform-li-version"
                  >
                    1.0.0
                  </span>
                </li>
                <li
                  class="cds--bmrg-aboutPlatform-li-between"
                >
                  <span
                    class="cds--bmrg-aboutPlatform-li"
                  >
                    Agents
                  </span>
                  <span
                    class="cds--bmrg-aboutPlatform-li-version"
                  >
                    1.0.0
                  </span>
                </li>
                <li
                  class="cds--bmrg-aboutPlatform-li-between-last"
                >
                  <span
                    class="cds--bmrg-aboutPlatform-li"
                  >
                    Scribeflow
                  </span>
                  <span
                    class="cds--bmrg-aboutPlatform-li-version"
                  >
                    1.0.0
                  </span>
                </li>
              </ul>
              <h1
                class="cds--bmrg-aboutPlatform-footer__header"
              >
                Copyright IBM Corp. 2022, 2025
              </h1>
            </div>
          </div>
        </div>
        <button
          class="cds--visually-hidden"
          type="button"
        >
          Focus sentinel
        </button>
      </div>
    </div>
  </div>
</body>
`;
